integrationTests/chainSimulator/rewards/rewards_test.go

```diff
@@ -88,50 +89,66 @@ func TestRewardsTxsAfterEquivalentMessages(t *testing.T) {
 	require.True(t, found)
 	require.NotNil(t, metaBlock)
 
-	nodesCoordinator := cs.GetNodeHandler(0).GetProcessComponents().NodesCoordinator()
+	coordinator := cs.GetNodeHandler(0).GetProcessComponents().NodesCoordinator()
 
-	shards := []uint32{0, 1, 2, core.MetachainShardId}
+	rewardsPerShard := computeRewardsForShards(metaBlock, coordinator, validators)
+
+	for shardID, reward := range rewardsPerShard {
+		fmt.Printf("rewards on shard %d: %s\n", shardID, reward.String())
+	}
+
+	require.True(t, allValuesEqual(rewardsPerShard))
```

Comment:
can you add a check for each reward tx value to be identical?
when every owner has one node, and all are eligible, with no missed blocks, we should have the same value distributed to each.